### PR TITLE
Freshontv search plugin

### PR DESCRIPTION
**Pull request by OmgOhnoes**
Adds a Freshon.tv search plugin.

Supports pagination (limited to 10 pages as users can set their pages to up to 100 items each).
Has a 100% and 50% torrent filter as well as HD and WebDL filters.

Downloads via RSS download link using the users passkey.

Note: throws a SSL error when getting the pages, not sure how to fix that as every recommendation doesn't seem to work including "verify_ssl_certificates: no"

### CODE REVIEW

File: flexget/plugins/search_freshon.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                'type': 'string',
+                'enum': ['free', 'half', 'all']
+            },
+        },
+        'required': ['username', 'password', 'passkey'],
+        'additionalProperties': False
+    }
+
+    def prepare_config(self, config):
+        config.setdefault('freeleech', 'all')
```
Comment (stevezau) [inline flexget/plugins/search_freshon.py]: You could use schema defaults here.

```
'category': {
    'type': 'string',
    'default': 'all',
    'enum': ['hd', 'webdl', 'all']
  },
```

File: flexget/plugins/search_freshon.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        if config['freeleech'] == 'all':
+            freeleech = 0
+        elif config['freeleech'] == 'free': 
+            freeleech = 3
+        elif config['freeleech'] == 'half': 
+            freeleech = 4
+        entries = set()
+
+        for search_string in entry.get('search_strings', [entry['title']]):
+            search_string = search_string.replace(' ', '-').lower()
```
Comment (stevezau) [inline flexget/plugins/search_freshon.py]: regex replace might be more efficient here.

File: flexget/plugins/search_freshon.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            search_string = search_string.replace('(', '')
+            search_string = search_string.replace(')', '')
+            query = normalize_unicode(search_string)
+            query_url_fragment = urllib.quote_plus(query.encode('utf-8'))
+            str_url = (base_url, search_fragment)
+            url = '/'.join(str_url) % (query_url_fragment, config['category'], str(freeleech))
+            log.debug('Search url: %s' % url)
+            f = session.get(url).content
```
Comment (stevezau) [inline flexget/plugins/search_freshon.py]: Could use task.session instead as it will solve your ssl issues

**Comment by OmgOhnoes** [comment]
I appreciate the feedback, still learning python and Flexget is a beast so i'm doing my best with previous plugin examples.

I've done some of the changes you've suggested and optimized the code a bit.

Using task.session does not seem to have fixed the SSL errors even while using "verify_ssl_certificates: no." On the plus side it seems to respect domain delays now.

**Comment by gazpachoking** [comment]
Thanks! Want to add a wiki page to searches section? http://flexget.com/wiki/Searches

**Comment by OmgOhnoes** [comment]
Will do, thanks!

**Pull request merged**